src/ACEtk/block/SecondaryParticleTypeBlock.hpp

```diff
@@ -82,9 +82,14 @@ class SecondaryParticleTypeBlock : protected details::ArrayData {
    */
   bool hasIP( unsigned int type ) const {
 
-    return std::find( this->begin(),
-                      this->begin() + this->NTYPE(),
-                      type ) != this->end();
+    return std::find_if(
+               this->begin(),
+               this->begin() + this->NTYPE(),
+               [type] ( auto&& value ) {
+
+                 return static_cast< unsigned int >( std::round( value ) )
```

Comment:
As an aside, I have found in our code that rounding to nearest (`std::round`) instead of rounding to zero `static_cast<int>` has a substantial performance penalty. However, our code has taken the stance that input is valid and was checked ahead of time for performance reasons, whereas this code cannot.

Reply:
ACEtk should not be used during simulations so performance is less of an issue - although it is a valid point I should look into.